src/main/java/de/torui/coflsky/minecraft_integration/CountdownTimer.java

```diff
@@ -0,0 +1,102 @@
+
+    public CountdownTimer() {
+    }
+
+    public static void onRenderTick(TickEvent.RenderTickEvent event) {
+        if (mc.currentScreen == null || mc.currentScreen instanceof GuiChat) // will only draw the Timer while no GUI or the Chat is open
+            if (currentEndTime - System.currentTimeMillis() > 0)
+                drawTimer();
+    }
+
+    /**
+     * @param seconds          will start a timer starting at seconds
+     * @param widthPercentage  width in correlation to the window size
+     * @param heightPercentage height in correlation to the window size
+     * @param fontScale        scales the text size by factor (1 = no change)
+     */
+    public static void startCountdown(double seconds, int widthPercentage, int heightPercentage, double fontScale) {
+        startCountdown(seconds, widthPercentage, heightPercentage, fontScale, "", 4);
+    }
+
+    /**
+     * @param seconds          will start a timer starting at seconds
+     * @param widthPercentage  width in correlation to the window size
+     * @param heightPercentage height in correlation to the window size
+     * @param fontScale        scales the text size by factor (1 = no change)
+     * @param prefix           will put that text infront of the seconds (supports color codes using §)
+     * @param maxPercision     length of the seconds in the timer
+     */
+    public static void startCountdown(double seconds, int widthPercentage, int heightPercentage, double fontScale, String prefix, int maxPercision) {
+		System.out.println("###Starting countdown " + seconds);
+        currentEndTime = (long) (System.currentTimeMillis() + (seconds * 1000));
+        currentWidth = widthPercentage;
+        currentHeight = heightPercentage;
+        currentScale = fontScale;
+        currentPrefix = prefix;
+        currentPercission = maxPercision;
+    }
+
+    public static void startCountdown(TimerData data) {
+        startCountdown(data.seconds, data.width, data.height, data.scale, data.prefix, data.maxPercision );
+    }
+
+    private static void drawTimer() {
+        long curMillis = currentEndTime - System.currentTimeMillis();
+        String render = getStringFromDouble(curMillis / 1000D);
+
+        ScaledResolution scaled = new ScaledResolution(mc);
+
+        GlStateManager.pushMatrix();
+        GlStateManager.scale(currentScale, currentScale, currentScale);
+        int scaledX = (int) (scaled.getScaledWidth() * (currentWidth / 100D)  / currentScale);
+        int scaledY = (int) (scaled.getScaledHeight() * (currentHeight / 100D)  / currentScale);
+        drawHVCenteredString(currentPrefix + render, scaledX, scaledY);
+        GlStateManager.popMatrix();
+    }
+
+    private static String getStringFromDouble(double seconds) {
+        String render;
+
+        if (seconds > 100) {
+            render = String.valueOf((int) seconds);
+        } else {
+            render = String.format(Locale.US, "%.3f", seconds).substring(0, currentPercission);
```

Comment:
Is this more efficient than e.g. `String.format(Locale.US, "%." + currenPrecision + "f", seconds)`?

Reply:
this is a special requirement to display different amount of precision based on the left time. 
It may be more efficient with a calculation but I couldn't get it to work as intended within half an hour.
123s, 34.3s and 2.45s